src/client/datascience/context/telemetry.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { Uri } from 'vscode';
+import { getOSType } from '../../common/utils/platform';
+import { EnvironmentType } from '../../pythonEnvironments/info';
+import { KernelConnectionMetadata } from '../jupyter/kernels/types';
+import * as hashjs from 'hash.js';
+import { Resource } from '../../common/types';
+import { IEventNamePropertyMapping, sendTelemetryEvent, sendTelemetryWhenDone } from '../../telemetry';
+import { StopWatch } from '../../common/utils/stopWatch';
+
+let connection: KernelConnectionMetadata;
+type ResourceSpecificTelemetryProperties = {
```

Comment:
This is sent with the telemetry (will move this into the telemetry file so we track what properties are sent & the like).